pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ProducerImpl.java

```diff
@@ -701,6 +816,59 @@ private void resendMessages(ClientCnx cnx) {
         });
     }
 
+    /**
+     * Removes checksum from {@link OpSendMsg} command if present else ignore it.   
+     * 
+     * @param op
+     */
+    private static void removeChecksum(OpSendMsg op) {
+        op.cmd.markReaderIndex();
+        int msgBufSize = op.cmd.readableBytes();
+        // ByteBuf can't use readBytes() to same buffer as it always requires readerIndex < writerIndex while writing
+        // into buffer. So, creating new temp ByteBuf to copy data without checksum.
+        ByteBuf temp = op.cmd.alloc().buffer(msgBufSize, msgBufSize);
```

Comment:
yes..removed copying part. However, at line [855](https://github.com/rdhabalia/pulsar/blob/7491a16c0f7f2e55ccce212e451447cca69b7776/pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ProducerImpl.java#L855) we are copying data-without-checksum to new ByteBuf,